ql/src/java/org/apache/hadoop/hive/ql/stats/ColStatsProcessor.java

```diff
@@ -127,15 +131,16 @@ public int process(Hive db, Table tbl) throws Exception {
             LOG.debug("Because {} is infinite or NaN, we skip stats.", columnName, e);
           }
         }
+        pos += columnStatsFields.size();
```

Comment:
all these methods are very wierd... because they just read 1 element at a time..but the position control is placed "outside" ... (here)

instead of starting a loop `for(int i=0;)` stuff; passing a columnindex iterator might make everything more readable...
if the iterator is not empty - a new column statistic clould be parsed from it

Reply:
Changed the method to use iterators.

Reply:
I want to circle back on this. Finally I had to undo this change since the exception above breaks the flow and you need to know the position in the outer method, it is not sufficient to rely on the iterator. I think the workflow could be slightly different if the exception would be caught in the internal methods; maybe worth tackling at some point.